app/gilded-rose.ts

```diff
@@ -17,53 +17,99 @@ export class GildedRose {
     this.items = items;
   }
 
-  updateQuality() {
-    for (let i = 0; i < this.items.length; i++) {
-      if (this.items[i].name != 'Aged Brie' && this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-        if (this.items[i].quality > 0) {
-          if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-            this.items[i].quality = this.items[i].quality - 1
-          }
-        }
-      } else {
-        if (this.items[i].quality < 50) {
-          this.items[i].quality = this.items[i].quality + 1
-          if (this.items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].sellIn < 11) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
-            }
-            if (this.items[i].sellIn < 6) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
-            }
-          }
-        }
-      }
-      if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-        this.items[i].sellIn = this.items[i].sellIn - 1;
+  static decrease(v : number, val : number = 1) : number {
+    // cannot decrease to lower than 0
+    return Math.max(v - val, 0)
+  }
+
+  static increase(v : number, val : number = 1) : number {
+    // cannot decrease to higher than 50
+    return Math.min(v + val, 50)
+  }
+
+  static ageBackstagePass(item : Item) : Item {
+    // Ages a backstage pass
+    if (item.sellIn > 10) {
+      item.quality = GildedRose.increase(item.quality)
+    }
+    else if (item.sellIn <= 10 && item.sellIn > 5) {
```

Comment:
The first half of the `&&` condition is unnecessary, since the previous if block covers that.